algorithms/src/std_algorithms/impl/Kokkos_RandomAccessIterator.hpp

```diff
@@ -155,6 +162,9 @@ class RandomAccessIterator< ::Kokkos::View<DataType, Args...> > {
  private:
   view_type m_view;
   ptrdiff_t m_current_index = 0;
+
+  template <class>
+  friend class RandomAccessIterator;
```

Comment:
since this is near the end of the class and far away from where it is actually needed, I think it would be helpful to add a comment here to say that this is needed fro the converting constructor accepting another iterator